Sources/TuistCore/Models/Project.swift

```diff
@@ -21,6 +21,7 @@ public struct Project: Hashable, Equatable, CustomStringConvertible, CustomDebug
     // MARK: - Attributes
 
     /// Path to the folder that contains the project manifest.
+    @available(*, deprecated, message: "The project path should be read from the graph.")
     public var path: AbsolutePath
```

Comment:
I appreciate the `graph` is used for traversing some of those properties during generation, but in isolation or prior to construction of the graph those properties (`path` and `targets`) are still needed on those models:

- During Manifest > Model conversion
- Within Model Mappers
- Constructing the graph itself

Perhaps this can be decoupled from this particular PR.

Reply:
The idea was to add it to prevent future contributions from using them. Otherwise, the moment we decide to remove them we might have way more usages than the ones that we currently have. I'll remove it from this PR.